homeassistant/components/hue/remote.py

```diff
@@ -0,0 +1,232 @@
+    ZLL_SWITCH_BUTTON_1_SHORT_RELEASED: "1_click_up",
+    ZLL_SWITCH_BUTTON_2_SHORT_RELEASED: "2_click_up",
+    ZLL_SWITCH_BUTTON_3_SHORT_RELEASED: "3_click_up",
+    ZLL_SWITCH_BUTTON_4_SHORT_RELEASED: "4_click_up",
+    ZLL_SWITCH_BUTTON_1_LONG_RELEASED: "1_hold_up",
+    ZLL_SWITCH_BUTTON_2_LONG_RELEASED: "2_hold_up",
+    ZLL_SWITCH_BUTTON_3_LONG_RELEASED: "3_hold_up",
+    ZLL_SWITCH_BUTTON_4_LONG_RELEASED: "4_hold_up",
+}
+TAP_BUTTONS = {
+    ZGP_SWITCH_BUTTON_1: "1_click",
+    ZGP_SWITCH_BUTTON_2: "2_click",
+    ZGP_SWITCH_BUTTON_3: "3_click",
+    ZGP_SWITCH_BUTTON_4: "4_click",
+}
+Z3_BUTTON = {
+    1000: "initial_press",
+    1001: "repeat",
+    1002: "short_release",
+    1003: "long_release",
+}
+Z3_DIAL = {1: "begin", 2: "end"}
+
+
+async def async_setup_entry(hass, config_entry: ConfigEntry, async_add_entities):
+    """Defer remote setup to the shared sensor module."""
+    await hass.data[HUE_DOMAIN][
+        config_entry.entry_id
+    ].sensor_manager.async_register_component("remote", async_add_entities)
+
+
+class HueGenericRemote(GenericHueSensor, RemoteDevice):
+    """Parent class to hold common Hue Remote entity info."""
+
+    @property
+    def icon(self):
+        """Icon to use in the frontend, if any."""
+        return REMOTE_ICONS.get(self.sensor.modelid[0:3])
+
+    def turn_on(self, **kwargs):
+        """Do nothing."""
+
+    def turn_off(self, **kwargs):
+        """Do nothing."""
+
+    def _parsed_sensor_last_updated(self):
+        """
+        Localize the 'lastupdated' timestamp attribute from sensor data.
+
+        Adding the bridge timestamp to state attributes ensures that at least
+        the last remote press is processed as a state change.
+        """
+        return dt_util.parse_datetime(self.sensor.lastupdated).astimezone(
+            self.hass.config.time_zone
+        )
+
+    @property
+    def device_state_attributes(self):
+        """Return the device state attributes."""
+        attributes = {
+            "model": self.sensor.type,
+            "last_button_event": self.state or "No data",
+            "last_updated": self._parsed_sensor_last_updated(),
+        }
+        if hasattr(self.sensor, "battery"):
+            attributes.update(
+                {
+                    "on": self.sensor.on,
+                    "reachable": self.sensor.reachable,
+                    "battery_level": self.sensor.battery,
+                }
+            )
+        return attributes
+
+
+class HueRemoteZLLSwitch(HueGenericRemote):
+    """
+    Class to hold ZLLSwitch remote entity info.
+
+    Models:
+    * RWL021, Hue Dimmer Switch
+    * ROM001, Hue Smart button
+    * Z3-1BRL, Lutron Aurora
+    """
+
+    @property
+    def state(self):
+        """Return the last button press of the remote."""
+        if self.sensor.modelid.startswith("RWL"):
+            return RWL_BUTTONS.get(self.sensor.state["buttonevent"])
+        return Z3_BUTTON.get(self.sensor.state["buttonevent"])
```

Comment:
We never represent remotes that send events to Home Assistant as entities, they are fired as events. This should not be an entity.